Mathlib/Combinatorics/Additive/Randomisation.lean

```diff
@@ -0,0 +1,50 @@
+/-
+Copyright (c) 2023 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import Mathlib.Analysis.Fourier.FiniteAbelian.Orthogonality
+import Mathlib.Combinatorics.Additive.Dissociation
+
+/-!
+# Randomising by a function of dissociated support
+
+This file proves that a function from a finite abelian group can be randomised by a function of
+dissociated support.
+
+Precisely, for `G` a finite abelian group and two functions `c : Dual G → ℝ` and `d : Dual G → ℝ`
```

Comment:
```suggestion
Precisely, for `G` a finite abelian group and two functions `c : AddChar G ℂ → ℝ` and `d : AddChar G ℂ → ℝ`
```
since the rest of this is basically valid lean